notifications/notifiers/base.py

```diff
@@ -0,0 +1,122 @@
+            user=notification_settings.user,
+            notification_type=notification_settings.notification_type,
+        )
+
+        if before is not None:
+            query = query.filter(created_on__lt=before.created_on)
+
+        return query.order_by('-created_on').first()
+
+    def can_notify(self, notification_settings, last_notification):
+        """
+        Returns true if we can notify this user based on their settings and when the last notification occurred
+
+        Args:
+            notification_settings (NotificationSettings): settings for this user and notification_type
+            last_notification (NotificationBase): last notification that was triggered for this NotificationSettings
+
+        Raises:
+            InvalidTriggerFrequencyError: if the frequency is invalid
+
+        Returns:
+            bool: True if we're due to send another notification
+        """
+        # special case if we've never sent a notification or the setting is for an immediate send
+        if last_notification is None or notification_settings.trigger_frequency == FREQUENCY_IMMEDIATE:
+            return True
+
+        # normalize now and created_at to the start of their respective days
+        normalized_now = normalize_to_start_of_day(now_in_utc())
+        normalized_create_at = normalize_to_start_of_day(last_notification.created_at)
+
+        if notification_settings.trigger_frequency == FREQUENCY_DAILY:
+            trigger_offset = DELTA_ONE_DAY
+        elif notification_settings.trigger_frequency == FREQUENCY_WEEKLY:
+            trigger_offset = DELTA_ONE_WEEK
+        else:
+            # practically, we'd only see this if our code called a notifier invalidly for a NEVER trigger_frequency
+            raise InvalidTriggerFrequencyError(
+                "Unsupported trigger_frequency: {}".format(notification_settings.trigger_frequency)
+            )
+
+        # return true if the normalized value is at least trigger_offset in the past
+        return (normalized_create_at + trigger_offset) <= normalized_now
+
+    def attempt_notify(self, notification_settings):
+        """
+        Attempts to notify the recipient
+
+        Args:
+            notification_settings (NotificationSettings): settings for this user and notification_type
+        """
+        last_notification = self._get_most_recent_notification(notification_settings)
+
+        if self.can_notify(notification_settings, last_notification):
+            self.notify(notification_settings, last_notification)
+
+    def notify(self, notification_settings, last_notification):  # pylint: disable=unused-argument
+        """
+        Attempts to notify the recipient via email
+
+        Args:
+            notification_settings (NotificationSettings): settings for this user and notification_type
+            last_notification (NotificationBase): last notification that was triggered for this NotificationSettings
```

Comment:
`last_notification` isn't used here, isn't passed as a parameter to `_create_notification`